crates/bindings-csharp/Runtime/Internal/ITable.cs

```diff
@@ -22,31 +22,51 @@ public class Enumerator(FFI.RowIter handle) : IDisposable
 
             public bool MoveNext()
             {
+                if (handle == FFI.RowIter.INVALID)
+                {
+                    return false;
+                }
+
                 uint buffer_len;
                 while (true)
                 {
                     buffer_len = (uint)buffer.Length;
-                    try
+                    var ret = (FFI.Errno)
+                        FFI._row_iter_bsatn_advance(handle, buffer, ref buffer_len);
+                    if (ret == FFI.Errno.EXHAUSTED)
                     {
-                        FFI._iter_advance(handle, buffer, ref buffer_len);
+                        handle = FFI.RowIter.INVALID;
                     }
-                    catch (BufferTooSmallException)
+                    switch (ret)
                     {
-                        buffer = new byte[buffer_len];
-                        continue;
+                        // Iterator exhausted, we're done.
+                        case FFI.Errno.EXHAUSTED:
+                        // Iterator advanced but not exhausted,
```

Comment:
I'm confused by this comment, is something missing? It's under `EXHAUSTED:` but says "iterator advanced but not exhausted". Also there is no `break;` for this `case` 🤔

Reply:
Ah yeah you can, in this case it was just unclear whether the fallthrough is intentional or not.

Nitpicking as the comment is enough but I'd probably just rewrite it to `case Errno.EXHAUSTED or Errno.OK:` like you said, which would make the intent clear instantly.